libpromises/policy.c

```diff
@@ -1425,6 +1430,16 @@ Constraint *PromiseAppendConstraint(Promise *pp, const char *lval, Rval rval, bo
         Constraint *old_cp = SeqAt(pp->conlist, i);
         if (strcmp(old_cp->lval, lval) == 0)
         {
+            if (strcmp(old_cp->lval, "ifvarclass") == 0 ||
+                strcmp(old_cp->lval, "if") == 0)
+            {
+                Buffer *grow = BufferNew();
+                BufferAppendF(grow, "(%s).", old_cp->rval);
+                BufferAppend(grow, RvalScalarValue(rval), strlen(RvalScalarValue(rval)));
```

Comment:
Because there's no ambiguity: `(x).y` is the same as `(x).(y)` for any `y`.  But I changed it anyway to be clear :)